lib/dns.js

```diff
@@ -133,10 +134,33 @@ function onlookupall(err, addresses) {
   }
 }
 
+function cleanExpiredDnsCache() {
+  const currentTime = DateNow();
+  for (const key in dnsCache) {
+    if (currentTime - dnsCache[key].timestamp >= 1000) {
+      delete dnsCache[key];
```

Comment:
Changing `dnsCache` to a `Map` could be a better/faster structure to use in this case, `delete` is usually slow.

Refs:
- https://stackoverflow.com/a/44008788
- https://github.com/RafaelGSS/nodejs-bench-operations/blob/9689895d6cc8e91b4598f1995c9e11eedf5d3a72/v20/RESULTS-v20_0_0.md#deleting-properties

Reply:
If we want to keep the object rather than a `Map`, just do a `dnsCache[key] = undefined`.